packages/compass-indexes/src/stores/store.ts

```diff
@@ -0,0 +1,116 @@
+import type { DataService } from 'mongodb-data-service';
+import type AppRegistry from 'hadron-app-registry';
+
+export type IndexesDataService = Pick<
+  DataService,
+  'indexes' | 'isConnected' | 'updateCollection' | 'createIndex' | 'dropIndex'
+>;
+export type ConfigureStoreOptions = {
+  dataProvider: {
+    dataProvider?: IndexesDataService;
+  };
+  namespace: string;
+  localAppRegistry: AppRegistry;
+  globalAppRegistry: AppRegistry;
+  serverVersion: string;
+  isReadonly: boolean;
+  isSearchIndexesSupported: boolean;
+};
+
+const configureStore = (options: ConfigureStoreOptions) => {
+  if (!options.dataProvider?.dataProvider) {
+    throw new Error(
+      "Can't configure store for indexes plugin without data serivce"
+    );
+  }
+  const store = createStore(
+    reducer,
+    {
+      dataService: options.dataProvider.dataProvider,
+      namespace: options.namespace,
+      serverVersion: options.serverVersion,
+      isReadonlyView: options.isReadonly,
+      searchIndexes: {
+        status: options.isSearchIndexesSupported
+          ? SearchIndexesStatuses.PENDING
+          : SearchIndexesStatuses.NOT_AVAILABLE,
+      },
+    },
+    applyMiddleware(thunk.withExtraArgument({}))
+  );
+
+  // Set the app registry if preset. This must happen first.
+  if (options.localAppRegistry) {
+    const localAppRegistry = options.localAppRegistry;
+    store.dispatch(localAppRegistryActivated(localAppRegistry));
+
+    localAppRegistry.on('refresh-data', () => {
+      void store.dispatch(fetchIndexes());
+    });
+
+    localAppRegistry.on(
+      'in-progress-indexes-added',
+      (index: InProgressIndex) => {
+        store.dispatch(inProgressIndexAdded(index));
+      }
+    );
+
+    localAppRegistry.on('in-progress-indexes-removed', (id: string) => {
+      store.dispatch(inProgressIndexRemoved(id));
+    });
+
+    localAppRegistry.on(
+      'in-progress-indexes-failed',
+      (data: { inProgressIndexId: string; error: string }) => {
+        store.dispatch(inProgressIndexFailed(data));
+      }
+    );
+  }
+
+  if (options.globalAppRegistry) {
+    const globalAppRegistry = options.globalAppRegistry;
+    store.dispatch(globalAppRegistryActivated(globalAppRegistry));
+
+    globalAppRegistry.on('refresh-data', () => {
+      void store.dispatch(fetchIndexes());
+    });
+
+    const instanceStore: any = globalAppRegistry.getStore('App.InstanceStore');
+    if (instanceStore) {
+      const instance = instanceStore.getState().instance;
+      // set the initial values
+      store.dispatch(writeStateChanged(instance.isWritable));
+      store.dispatch(getDescription(instance.description));
+
+      // these can change later
+      instance.on('change:isWritable', () => {
+        store.dispatch(writeStateChanged(instance.isWritable));
+      });
+      instance.on('change:description', () => {
+        store.dispatch(getDescription(instance.description));
+      });
+    }
+  }
+
+  void store.dispatch(fetchIndexes());
```

Comment:
This probably should be triggered from the UI, but keeping it here as was.

Reply:
Yeah, when you are doing similar for search indexes, would probably be good to move this to UI too